tests/test_view.py

```diff
@@ -59,4 +63,238 @@ def test_pycon(self):
         self.assertFalse(view.mgr.pycon.python_redirect)
 
 
+class ViewCameraTB:
+    camera_type = None
+
+    @classmethod
+    def setUpClass(cls):
+        widget = view.RManager.instance.setUp().add3DWidget()
+        widget.setCameraType(cls.camera_type)
+
+        cls.widget = widget
+        cls.controller = widget.cameraController()
+
+        cls.move = cls.controller.updateCameraPosition
+
+        cls.pos = cls.controller.position
+        cls.look_speed = cls.controller.lookSpeed
+        cls.linear_speed = cls.controller.linearSpeed
+        cls.view_vector = cls.controller.viewVector
+        cls.view_center = cls.controller.viewCenter
+        cls.up_vector = cls.controller.upVector
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.widget.close_and_destroy()
+
+    def angle_axis(self, angle_deg, axis):
+        a = axis
+        angle = np.radians(angle_deg)
+
+        cos_angle = np.cos(angle)
+        sin_angle = np.sin(angle)
+
+        rotation = np.identity(3) * cos_angle
+
+        m = np.outer(a, a) * (1 - cos_angle)
+        n = np.array([
+            [0, -a[2], a[1]],
+            [a[2], 0, -a[0]],
+            [-a[1], a[0], 0]
+        ]) * sin_angle
+
+        rotation += m + n
+
+        return rotation
+
+    def normalize(self, vec):
+        return vec / np.linalg.norm(vec)
+
+
+@unittest.skipIf(GITHUB_ACTIONS, "GUI is not available in GitHub Actions")
+class ViewFPSCameraTC(ViewCameraTB, unittest.TestCase):
+    camera_type = "fps"
+
+    def setUp(self):
+        self.widget.resetCamera()
+
+    def test_reset(self):
+        dt = 0.01
+        initial_pos = self.pos()
+        initial_view_vector = self.view_vector()
+        initial_view_center = self.view_center()
+        initial_up_vector = self.up_vector()
+
+        self.move(x=1, y=1, z=1, dt=dt)
+        self.move(yaw=1, pitch=1, dt=dt, left_mouse_button=True)
+
+        self.assertNotEqual(self.pos(), initial_pos)
+        self.assertNotEqual(self.view_vector(), initial_view_vector)
+        self.assertNotEqual(self.view_center(), initial_view_center)
+        self.assertNotEqual(self.up_vector(), initial_up_vector)
+
+        self.widget.resetCamera()
+
+        self.assertEqual(self.pos(), initial_pos)
+        self.assertEqual(self.view_vector(), initial_view_vector)
+        self.assertEqual(self.view_center(), initial_view_center)
+        self.assertEqual(self.up_vector(), initial_up_vector)
+
+    def test_translation(self):
+        dt = 0.01
+        delta = dt * self.linear_speed()
+        delta_vec = np.array([delta, delta, -delta])
+        new_pos = np.array(self.pos()) + delta_vec
+        new_view_center = np.array(self.view_center()) + delta_vec
+
+        self.move(x=1, dt=dt)
+        self.assertEqual(self.pos()[0], new_pos[0])
```

Comment:
Exact comparison tells a lot.  This is great.

It is a productive practice for the author of a code review (pull request) to add review comments at the place that worth reading.  The author is the most familiar with the code and they can make the comments in the fastest way.  Without the annotation it will take a review more time to understand the code.

Reply:
Thank you for the feedback) Will take into account